spec/remote_command_runner_spec.rb

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2013-2014 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+require "spec"
+
+describe RemoteCommandRunner do
+  let(:ssh_output) { "-rw-r--r-- 1 root root 642 Sep 27 22:06 /etc/hosts" }
+  subject { RemoteCommandRunner.new("1.2.3.4") }
+
+  describe "#run" do
+    it "calls ssh and returns the command's standard output" do
+      expect(Cheetah).to receive(:run).
+        and_return(ssh_output)
+
+      output = subject.run("ls", "-l", "/etc/hosts", {:stdout=>:capture})
```

Comment:
Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Surrounding space missing for operator '=>'.<br>Space inside { missing.<br>Space inside } missing.